File: TradingSystem/Store/Application/OrderDTO.cs
namespace Store.Application;

public class OrderDTO
{
    protected long orderId;
    protected int amount;
    protected ProductSupplierDTO productSupplier = new();

    public long OrderId
    {
        get => orderId;
        set => orderId = value;
    }
    
    public int Amount
    {
        get => amount;
        set => amount = value;
    }

    public ProductSupplierDTO ProductSupplier
    {
        get => productSupplier;
        set => productSupplier = value ?? throw new ArgumentNullException(nameof(value));
    }
}

File: TradingSystem/WebServerStore/StoreApp/src/app/classes/OrderDTO.ts
import { ProductSupplierDTO } from "./ProductSupplierDTO";

export class OrderDTO {
    orderId: number;
    amount : number;
    productSupplier: ProductSupplierDTO;

    constructor(
        orderId: number,
        amount : number,
        productSupplier: ProductSupplierDTO) {
            this.orderId = orderId;
            this.amount = amount;
            this.productSupplier = productSupplier;
    }
}